File: HslCommunication_Java/HslCommunication/src/HslCommunication/Core/Address/DeviceAddressBase.java
package HslCommunication.Core.Address;

/**
 * 所有通信类的地址基类
 */
public class DeviceAddressBase {


    /**
     * 获取地址信息
     * @return
     */
    public int getAddress() {
        return Address;
    }

    /**
     * 设置地址信息
     * @param address
     */
    public void setAddress(int address) {
        Address = address;
    }

    /**
     * 起始地址
     */
    private int Address = 0;


    /**
     * 解析字符串的地址
     *
     * @param address
     */
    public void AnalysisAddress(String address) {
        Address = Integer.parseInt(address);
    }


    /**
     * 返回表示当前对象的字符串
     * @return 字符串
     */
    @Override
    public String toString() {
        return String.valueOf(Address);
    }
}

File: HslCommunication_Net45/Core/Address/DeviceAddressBase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;







namespace HslCommunication.Core.Address
{
    /// <summary>
    /// 所有设备通信类的地址基础类
    /// </summary>
    public class DeviceAddressBase
    {
        /// <summary>
        /// 起始地址
        /// </summary>
        public ushort Address { get; set; }


        /// <summary>
        /// 解析字符串的地址
        /// </summary>
        /// <param name="address">地址信息</param>
        public virtual void AnalysisAddress( string address )
        {
            Address = ushort.Parse( address );
        }


        /// <summary>
        /// 返回表示当前对象的字符串
        /// </summary>
        /// <returns>字符串数据</returns>
        public override string ToString( )
        {
            return Address.ToString( );
        }

    }
}
